cassh

```diff
@@ -121,10 +121,9 @@ class CASSH(object):
             if prefix is not None:
                 auth.update(prefix)
             return '?%s' % urlencode(auth)
-        else:
-            if prefix is None:
-                return ''
-            return '?%s' % urlencode(prefix)
+        if prefix is None:
```

Comment:
Euh ok, maybe the 2 `prefix` checks could be grouped in a easier `if else` statement ?
And I think the description of the method is not right, is it ?

If it is I do not understand :p

Reply:
the problem is the return. A best practice is to not use a "else" if there is already a "return".
```
cassh:119:8: R1705: Unnecessary "else" after "return" (no-else-return)
```

Reply:
I agree with you about the unnecessary "else" after "return", but I was refering the 2 `if prefix` tests.

Could if be possible to do something like this ?


```python
query_string = {}
...

if self.auth == 'ldap':
    auth = {'realname': self.realname, 'password': getpass(passwd_message)}
    query_string.update(auth)

if prefix is not None:
    query_string.update(prefix)

return '?%s' % urlencode(query_string)
```

Reply:
We have to do more tests because I'm not sure it is working. You skip the case where there is no ldap and no prefix.
I'll try it tomorrow.

Reply:
When there are no LDAP and Prefix, then `query_string` is an empty dict ... so `urlencode(query_string)` should be ok

Anyway you are right, we have to test :)